graph_service_client.go

```diff
@@ -1,854 +1,43 @@
-}
-// TeamsTemplates the teamsTemplates property
-func (m *GraphServiceClient) TeamsTemplates()(*i1b75be7b5675627960b4672ab148be21ff379d5cbc0e62f6bc5b97d54464f8b5.TeamsTemplatesRequestBuilder) {
-    return i1b75be7b5675627960b4672ab148be21ff379d5cbc0e62f6bc5b97d54464f8b5.NewTeamsTemplatesRequestBuilderInternal(m.pathParameters, m.requestAdapter);
-}
-// TeamsTemplatesById gets an item from the github.com/microsoftgraph/msgraph-sdk-go/.teamsTemplates.item collection
-func (m *GraphServiceClient) TeamsTemplatesById(id string)(*i2a87b2a2ef0c6367c3abc2e909d9176a42e3124d1896753ad0a25d9a6a881c32.TeamsTemplateItemRequestBuilder) {
-    urlTplParams := make(map[string]string)
-    for idx, item := range m.pathParameters {
-        urlTplParams[idx] = item
-    }
-    if id != "" {
-        urlTplParams["teamsTemplate%2Did"] = id
-    }
-    return i2a87b2a2ef0c6367c3abc2e909d9176a42e3124d1896753ad0a25d9a6a881c32.NewTeamsTemplateItemRequestBuilderInternal(urlTplParams, m.requestAdapter);
-}
-// Teamwork the teamwork property
-func (m *GraphServiceClient) Teamwork()(*ie05ac24b652f7d895cca374316c093c4ca40dd2df0f1518c465233d6432b1ef9.TeamworkRequestBuilder) {
-    return ie05ac24b652f7d895cca374316c093c4ca40dd2df0f1518c465233d6432b1ef9.NewTeamworkRequestBuilderInternal(m.pathParameters, m.requestAdapter);
-}
-// Users the users property
-func (m *GraphServiceClient) Users()(*if6ffd1464db2d9c22e351b03e4c00ebd24a5353cd70ffb7f56cfad1c3ceec329.UsersRequestBuilder) {
-    return if6ffd1464db2d9c22e351b03e4c00ebd24a5353cd70ffb7f56cfad1c3ceec329.NewUsersRequestBuilderInternal(m.pathParameters, m.requestAdapter);
-}
-// UsersById gets an item from the github.com/microsoftgraph/msgraph-sdk-go/.users.item collection
-func (m *GraphServiceClient) UsersById(id string)(*i009581390843c78f63b06f9dcefeeb5ef2a124a2ac1dcbad3adbe4d0d5e650af.UserItemRequestBuilder) {
-    urlTplParams := make(map[string]string)
-    for idx, item := range m.pathParameters {
-        urlTplParams[idx] = item
-    }
-    if id != "" {
-        urlTplParams["user%2Did"] = id
-    }
-    return i009581390843c78f63b06f9dcefeeb5ef2a124a2ac1dcbad3adbe4d0d5e650af.NewUserItemRequestBuilderInternal(urlTplParams, m.requestAdapter);
+
+func NewGraphClient(adapter abstractions.RequestAdapter) *GraphServiceClient {
+	client := NewGraphServiceClient(adapter)
+	return &GraphServiceClient{
+		client,
+	}
 }
-// Workbooks the workbooks property
-func (m *GraphServiceClient) Workbooks()(*i088cccaaaff488138f258ec600ca804c5dc9548772ebe52bf3cb7a3eaf4b9fdf.WorkbooksRequestBuilder) {
-    return i088cccaaaff488138f258ec600ca804c5dc9548772ebe52bf3cb7a3eaf4b9fdf.NewWorkbooksRequestBuilderInternal(m.pathParameters, m.requestAdapter);
+
+// NewGraphServiceClientWithCredentials instantiates a new GraphBaseServiceClient with provided credentials and scope
+func NewGraphServiceClientWithCredentials(credential azcore.TokenCredential, scopes []string) (*GraphServiceClient, error) {
```

Comment:
we need an overload with the valid hosts as a parameter as well.

Reply:
@baywet Just so I'm clear are we talking about the host url thats passed here `adapter.SetBaseUrl(host)`?

Reply:
no, the ones passed to the [authentication provider here](https://github.com/microsoft/kiota-authentication-azure-go/blob/d842601fac75dde1f3ed97ba097815521d1ccb13/azure_identity_authentication_provider.go#L26) and defaulted [here](https://github.com/microsoft/kiota-authentication-azure-go/blob/d842601fac75dde1f3ed97ba097815521d1ccb13/azure_identity_access_token_provider.go#L47). Which would effectively solve [this issue](https://github.com/microsoft/kiota/issues/1063 for Go)